module/window_swallowing.lua

```diff
@@ -13,20 +13,51 @@ local window_swallowing_activated = false
 -- you might want to add or remove applications here
 local dont_swallow_classname_list = beautiful.dont_swallow_classname_list
     or { "firefox", "Gimp", "Google-chrome" }
+local cant_swallow_classname_list = beautiful.cant_swallow_classname_list
+    or { "Yad" }
 local activate_dont_swallow_filter = beautiful.dont_swallow_filter_activated
     or true
 
--- checks if client classname matches with any entry of the dont-swallow-list
-local function check_if_swallow(c)
+-- check if element exist in table
+local function is_in_Table(element, table)
+    local res = false
+    for _, value in pairs(table) do
+        if element:match(value) then
+            res = true
+            break
+        end
+    end
+    return res
+end
+
+-- checks if client classname can be swallowed
+local function check_if_swallow(class)
+    local res
     if not activate_dont_swallow_filter then
-        return true
+        res = true
+    else
+        if is_in_Table(class, dont_swallow_classname_list) then
+            res = false
+        else
+            res = true
+        end
     end
-    for _, classname in ipairs(dont_swallow_classname_list) do
-        if classname == c.class then
-            return false
+    return res
+end
+
+-- checks if client classname can swallow it's parent
+local function check_can_swallow(class)
+    local res
+    if not activate_dont_swallow_filter then
+        res = true
+    else
+        if is_in_Table(class, cant_swallow_classname_list) then
+            res = false
```

Comment:
Just return `not is_in_Table(class, cant_swallow_classname_list)` here already and get rid of the `res` variable. Same for the `check_if_swallow` function. Readability won't suffer from that

Reply:
welp i did this since i usually don't like having multiple return statements on a function as i usually find it harder to debug but if this is the style for bling then i will push a commit to remove the res var on the functions.